src/copyright/ui/ajax-copyright-hist.php

```diff
@@ -226,10 +227,14 @@ private function getTableData($upload, $item, $agent_pk, $type, $listPage, $filt
    * @param boolean $activated           True to get activated copyrights, else false
    * @return array[][] Array of table records, filtered records, total records
    */
-  protected function getCopyrights($upload_pk, $item, $uploadTreeTableName, $agentId, $type, $filter, $activated = true)
+  public function getCopyrights($upload_pk, $item, $uploadTreeTableName, $agentId, $type, $filter, $activated = true, $offset = null , $limit = null)
   {
-    $offset = GetParm('iDisplayStart', PARM_INTEGER);
-    $limit = GetParm('iDisplayLength', PARM_INTEGER);
+    if($offset==null){
+      $offset = GetParm('iDisplayStart', PARM_INTEGER);
+    }
+    if ($limit==null){
+      $limit = GetParm('iDisplayLength', PARM_INTEGER);
+    }
```

Comment:
Would instead recommend to update the function call on line 206 to send the limit and offset reading them from `GetParm()` and do that here if they are null.

Reply:
So there would be two instances of GetParm() right? One for the function call in Line 206 and one inside the function for the case of null parameters right?

Reply:
```php
$offset = GetParm('iDisplayStart', PARM_INTEGER);
$limit = GetParm('iDisplayLength', PARM_INTEGER);
list ($rows, $iTotalDisplayRecords, $iTotalRecords) = $this->getCopyrights($upload, $item, $this->uploadtree_tablename, $agent_pk, $type, $filter, $activated, $offset, $limit);
```

Reply:
No need to check for `null` inside the `getCopyrights()`

Reply:
got it thank you!!